src/app/projects/projects.component.ts

```diff
@@ -117,19 +117,30 @@ export class ProjectsComponent implements OnInit, OnDestroy {
               this.isLoading = false;
             })
             .subscribe((projects: Project[]) => {
-              // Get all shapefiles.
               this.documentService.getAll('SHAPEFILE')
               .takeUntil(this.ngUnsubscribe)
               .subscribe(shapefiles => {
                   // "Attach" a shapefile to its project.
                   if (shapefiles) {
                     projects.forEach(project => {
-                      // project.shapefiles = [];
+                      if (Array.isArray(project?.shapefiles) && project?.shapefiles.length > 0) {
+                        return;
+                      }
+                      /**
+                       * If no shapefiles are attached, try to find another that's
```

Comment:
It looks like this section allows for legacy shapefile functionality. If so, that's great.